<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>300</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>300</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>300</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QLabel" name="label">
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:36pt;&quot;&gt;CW Tools&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="minimumSize">
       <size>
        <width>280</width>
        <height>430</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>280</width>
        <height>430</height>
       </size>
      </property>
      <property name="toolTip">
       <string notr="true"/>
      </property>
      <property name="locale">
       <locale language="English" country="Canada"/>
      </property>
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Utility</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPushButton" name="btn_buildZeroGrp">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="locale">
           <locale language="English" country="Canada"/>
          </property>
          <property name="text">
           <string>Build zero group</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btn_placeLocatorAtVertCenter">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="locale">
           <locale language="English" country="Canada"/>
          </property>
          <property name="text">
           <string>Vertices centered locator</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btn_renameShapeNodes">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="locale">
           <locale language="English" country="Canada"/>
          </property>
          <property name="text">
           <string>Rename shape nodes</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btn_setHistoryNotInteresting">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="locale">
           <locale language="English" country="Canada"/>
          </property>
          <property name="text">
           <string>Set history not interesting</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btn_sortSelection">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="locale">
           <locale language="English" country="Canada"/>
          </property>
          <property name="text">
           <string>Sort selection in outliner</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btn_mergeCrv">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="locale">
           <locale language="English" country="Canada"/>
          </property>
          <property name="text">
           <string>Merge curves</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Rigging</string>
       </attribute>
      </widget>
      <widget class="QWidget" name="tab_ctrl">
       <attribute name="title">
        <string>Control</string>
       </attribute>
       <widget class="QWidget" name="wgt_colorPicker" native="true">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>91</y>
          <width>250</width>
          <height>181</height>
         </rect>
        </property>
        <property name="minimumSize">
         <size>
          <width>250</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>250</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="autoFillBackground">
         <bool>true</bool>
        </property>
        <property name="styleSheet">
         <string notr="true">background{rgb(0,0,0)}</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="3" column="0" colspan="3">
          <widget class="QPushButton" name="btn_setCtrlColor">
           <property name="text">
            <string>Set color</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="lbl_colorIndicator">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>500</width>
             <height>500</height>
            </size>
           </property>
           <property name="baseSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="layoutDirection">
            <enum>Qt::LayoutDirection::LeftToRight</enum>
           </property>
           <property name="autoFillBackground">
            <bool>false</bool>
           </property>
           <property name="styleSheet">
            <string notr="true">background-color: rgb(0, 0, 0); border: 1px solid #888;</string>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="7" column="0" colspan="3">
          <widget class="QPushButton" name="btn_setCtrlWidth">
           <property name="text">
            <string>Set width</string>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLineEdit" name="txt_width">
           <property name="minimumSize">
            <size>
             <width>90</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>90</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="inputMask">
            <string notr="true"/>
           </property>
           <property name="text">
            <string>1</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QSlider" name="sdr_width">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>50</number>
           </property>
           <property name="sliderPosition">
            <number>1</number>
           </property>
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="lbl_color">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string>Color</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
           </property>
           <property name="margin">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSlider" name="sdr_color">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximum">
            <number>31</number>
           </property>
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="lbl_width">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string>Width</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="widget_2" native="true">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>12</y>
          <width>250</width>
          <height>79</height>
         </rect>
        </property>
        <property name="minimumSize">
         <size>
          <width>250</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>250</width>
          <height>16777215</height>
         </size>
        </property>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="lbl_ctrlName">
           <property name="minimumSize">
            <size>
             <width>90</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>90</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string>Control name</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="txt_ctrlName"/>
         </item>
         <item row="1" column="0" colspan="2">
          <widget class="QPushButton" name="btn_extractShape">
           <property name="text">
            <string>Add curve to library</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="widget_3" native="true">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>282</y>
          <width>250</width>
          <height>84</height>
         </rect>
        </property>
        <property name="minimumSize">
         <size>
          <width>250</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>250</width>
          <height>16777215</height>
         </size>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="1">
          <widget class="QCheckBox" name="cbx_ctrlZeroGrp">
           <property name="text">
            <string>add zero grp</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0" colspan="2">
          <widget class="QPushButton" name="btn_createShape">
           <property name="text">
            <string>Create ctrl shape</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QComboBox" name="cbb_setCtrlShape"/>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="btn_close">
      <property name="minimumSize">
       <size>
        <width>280</width>
        <height>50</height>
       </size>
      </property>
      <property name="locale">
       <locale language="English" country="Canada"/>
      </property>
      <property name="text">
       <string>close</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>sdr_width</sender>
   <signal>sliderReleased()</signal>
   <receiver>txt_width</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>218</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>91</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txt_width</sender>
   <signal>editingFinished()</signal>
   <receiver>sdr_width</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>91</x>
     <y>327</y>
    </hint>
    <hint type="destinationlabel">
     <x>218</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
